Automattic/jetpack
# Composer Packages: change the package type to jetpack-library

#### Changes proposed in this Pull Request:
* Strings in packages that are installed in the 'vendor' directory are not translated. Therefore, the Jetpack Composer packages should be installed in a different directory. The new `jetpack-composer-plugin` package will install packages with the `jetpack-library` type in the `jetpack_vendor` directory. Update the type for each of the Jetpack Composer packages to `jetpack-library`.
* Use the new `jetpack-composer-plugin` package in the Jetpack plugin.

As mentioned in [this comment](https://github.com/Automattic/jetpack/pull/21877#pullrequestreview-819166602), it should be sufficient to have the plugins depend on the `jetpack-composer-plugin` package.

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:
Here's how I tested the new custom Composer installer. If you have a better approach, please let me know.
1. Checkout this branch.
2. Clean the Jetpack plugin's `vendor` folder using the `jetpack clean` command and following the prompts.
3. In the `jetpack/projects/plugins/jetpack` directory, run `composer update automattic/jetpack-composer-plugin`.
4. Inspect the `jetpack/projects/plugins/jetpack/vendor/automattic` directory. It should contain only the `jetpack-autoloader`, `jetpack-changelogger`, and `jetpack-composer-plugin` packages.
5. Inspect the `jetpack/projects/plugins/jetpack/jetpack-vendor` directory. It should contain 30 `automattic/jetpack-*` packages.
6. Smoke test Jetpack to make sure that the package files are loaded correctly from the new `jetpack_vendor` directory. Install and activate Jetpack. Connect to WPCOM and confirm that Jetpack is working.
